### PR TITLE
Fix `alicloud_cdn_domain_config` a bug that need to be deleted when updating

### PR DESCRIPTION
**Pull request by xinmu0426**
Fix `alicloud_cdn_domain_config` a bug that need to be deleted when updating

### CODE REVIEW

File: alicloud/resource_alicloud_cdn_domain_config.go [hunk 1]
```diff
@@ -33,8 +37,19 @@ func resourceAlicloudCdnDomainConfig() *schema.Resource {
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cdn_domain_config.go:50]: config_id and status should be computed.

File: alicloud/service_alicloud_cdn.go [hunk 2]
```diff
@@ -169,3 +199,24 @@ func (c *CdnService) DescribeTags(resourceId string, resourceType TagResourceTyp
...
+		}
+
+		val := v.(map[string]interface{})
+		for _, failState := range failStates {
+			if fmt.Sprint(val["Status"]) == failState {
+				return val, fmt.Sprint(val["Status"]), WrapError(Error(FailedToReachTargetStatus, fmt.Sprint(val["Status"])))
+			}
+		}
+		return val, fmt.Sprint(val["Status"]), nil
+	}
```
Comment (xiaozhu36) [inline alicloud/service_alicloud_cdn.go:221]: Please update the docs

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_cdn_domain_config.go [hunk 3]
```diff
@@ -96,11 +137,96 @@ func resourceAlicloudCdnDomainConfigCreate(d *schema.ResourceData, meta interfac
...
+					return resource.RetryableError(err)
+				}
+				return resource.NonRetryableError(err)
+			}
+			return nil
+		})
+
+		addDebug(action, response, request)
+		v, err := jsonpath.Get("$.DomainConfigList.DomainConfigModel", response)
+		if err != nil {
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_cdn_domain_config.go:203]: 放置debug后,json解析前

File: alicloud/resource_alicloud_cdn_domain_config.go [hunk 4]
```diff
@@ -109,63 +235,92 @@ func resourceAlicloudCdnDomainConfigRead(d *schema.ResourceData, meta interface{
 		return WrapError(err)
 	}
 
+	val := v.(map[string]interface{})
+	args := val["FunctionArgs"].(map[string]interface{})["FunctionArg"].([]interface{})
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_cdn_domain_config.go:239]: 连续强转最好处理一下

File: alicloud/resource_alicloud_cdn_domain_config_test.go [hunk 5]
```diff
@@ -11,6 +11,92 @@ import (
...
+
+	rac := resourceAttrCheckInit(rc, ra)
+
+	testAccCheck := rac.resourceAttrMapUpdateSet()
+	rand := acctest.RandIntRange(1000000, 9999999)
+	name := fmt.Sprintf("tf-testacc%s%d.xiaozhu.com", defaultRegionToTest, rand)
+	testAccConfig := resourceTestAccConfigFunc(resourceId, name, resourceCdnDomainConfigDependence)
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_cdn_domain_config_test.go:33]: 有没有CI跑不了的Region?

Comment (xinmu0426) [reply alicloud/resource_alicloud_cdn_domain_config_test.go:33]: 都可以跑的

File: alicloud/resource_alicloud_cdn_domain_config.go [hunk 6]
```diff
@@ -75,19 +91,44 @@ func resourceAlicloudCdnDomainConfigCreate(d *schema.ResourceData, meta interfac
...
 	})
+
+	addDebug(action, response, request)
+	v, err := jsonpath.Get("$.DomainConfigList.DomainConfigModel", response)
 	if err != nil {
-		return WrapErrorf(err, DefaultErrorMsg, "cdn_domain_config", request.GetActionName(), AlibabaCloudSdkGoERROR)
+		return WrapErrorf(err, FailedGetAttributeMsg, "request.DomainNames", "$.DomainConfigList.DomainConfigModel", err)
+	}
+
+	if len(v.([]interface{})) != 1 {
```
[the remark this answers is not in the document]
Comment (xinmu0426) [reply alicloud/resource_alicloud_cdn_domain_config.go:121]: 每次只能配置一条config，创建后大于或者等于1都不可以

**Pull request merged**